plugins/usbdmx/ShowJockey.cpp

```diff
@@ -0,0 +1,393 @@
+                           libusb_device_handle *usb_handle,
+                           int max_packet_size_out,
+                           int endpoint);
+
+ private:
+  LibUsbAdaptor* const m_adaptor;
+  int m_max_packet_size_out;
+  int m_endpoint;
+
+  bool TransmitBuffer(libusb_device_handle *handle,
+                      const DmxBuffer &buffer);
+  int bulkSync(libusb_device_handle *handle,
+               int endpoint,
+               int max_packet_size,
+               unsigned char *buffer,
+               int size);
+};
+
+ShowJockeyThreadedSender::ShowJockeyThreadedSender(
+    LibUsbAdaptor *adaptor,
+    libusb_device *usb_device,
+    libusb_device_handle *usb_handle,
+    int max_packet_size_out,
+    int endpoint)
+    : ThreadedUsbSender(usb_device, usb_handle),
+      m_adaptor(adaptor),
+      m_max_packet_size_out(max_packet_size_out),
+      m_endpoint(endpoint) {
+}
+
+bool ShowJockeyThreadedSender::TransmitBuffer(libusb_device_handle *handle,
+                                              const DmxBuffer &buffer) {
+  if (!handle) {
+    return false;
+  }
+
+  int ret_val = 0;
+  int left_write_size = 512;
+  uint16_t already_written_size = 0;
+  unsigned int write_size = 0;
+  int max_packet_size_out = m_max_packet_size_out;
+
+  unsigned char *bulk_buffer = new unsigned char[max_packet_size_out]();
+
+  int bulkDataSize = max_packet_size_out - 2;
+  unsigned int slot = 0;
+
+  while (left_write_size > 0) {
+    memset(bulk_buffer, 0, max_packet_size_out);
+    memcpy(bulk_buffer, &already_written_size, 2);
+
+    if (left_write_size > bulkDataSize) {
+      write_size = bulkDataSize;
+    } else {
+      write_size = left_write_size;
+    }
+
+    buffer.GetRange(slot, bulk_buffer + 2, &write_size);
+
+    ret_val = bulkSync(handle, m_endpoint, max_packet_size_out,
+                      bulk_buffer, write_size + 2);
+
+    if (ret_val < 0) {
+      return ret_val;
+    }
+
+    left_write_size -= write_size;
+    already_written_size += write_size;
+    slot += write_size;
+  }
+
+  if (bulk_buffer) {
+    delete[] bulk_buffer;
+    bulk_buffer = 0;
+  }
+
+  return true;
+}
+
+int ShowJockeyThreadedSender::bulkSync(libusb_device_handle *handle,
+                                       int endpoint, int max_packet_size,
+                                       unsigned char *buffer, int size) {
+  if (!handle) {
+    return -1;
+  }
+
+  if (size <= max_packet_size) {
+    int transferred = -1;
+    m_adaptor->BulkTransfer(handle, endpoint, buffer, size,
+                            &transferred, URB_TIMEOUT_MS);
+
+    return transferred;
+  } else {
+    int ret_val = 1;
+    unsigned char *buf = buffer;
+    int n_packet = max_packet_size;
+    int n_left = size;
+    int transferred = -1;
+    int transferredAll = 0;
+
+    while (1) {
+      if (n_left <= 0) {
+        break;
+      }
+
+      ret_val = m_adaptor->BulkTransfer(handle, endpoint, buffer, n_packet,
```

Comment:
In fact, from what I tested, this part of the code is never reach. The bulk transfer of the line 192 is enough to transmit the data. I kept those line because they part of the driver developed by ShowJockey team I am trying to adapt to Ola.
I can remove them if you want.